src/main/java/nextstep/qna/domain/Answers.java

```diff
@@ -0,0 +1,33 @@
+package nextstep.qna.domain;
+
+import java.util.ArrayList;
+import java.util.List;
+import nextstep.users.domain.NsUser;
+
+public class Answers {
+
+    private final List<Answer> answers;
+
+    public Answers() {
+        this.answers = new ArrayList<>();
+    }
+
+    public void add(Answer answer) {
+        this.answers.add(answer);
+    }
+
+    public boolean canBeDeletedBy(NsUser user) {
+        return answers.stream().allMatch(answer -> answer.isOwner(user));
+    }
+
+    public List<DeleteHistory> deleteAll(NsUser user) {
+        List<DeleteHistory> deleteHistories = new ArrayList<>();
+        for (Answer answer : answers) {
+            if (answer.isOwner(user)) {
+                answer.setDeleted(true);
+                deleteHistories.add(DeleteHistory.ofAnswer(answer));
+            }
+        }
+        return deleteHistories;
```

Comment:
```suggestion
    public void delete(NsUser writer) {
        for (Answer answer : answers) {
            answer.delete(writer);
        }
    }
    public void delete(NsUser writer, DeleteHistories deleteHistories) {
        this.delete(writer);
        if (deleteHistories != null) {
            deleteHistories.addAllAnswers(answers);
        }
    }
```

- 이전 미션의 프로그래밍 요구사항도 모주 준수하셔야 해요. (indent는 1까지 허용)
- 이력 저장을 위한 일급컬렉션을 도출하고 역할을 위임할 수 있어요
- 권한을 확인하고, soft delete 하는 책임을 Answer 객체에 위임할 수 있어요.